Cleaner formatting for trading charts date axes

deusmax opened the pull request:
Clean up date axes style for easy reading. Made date labels shorter,
with clear visual correspondence to relevant axes tick mark.
Tick marks with labels are now larger than those without.

<!-- 
- make yourself familiar with the CONTRIBUTING.md if you have not already (https://github.com/bisq-network/bisq/blob/master/CONTRIBUTING.md)
- make sure you follow our [coding style guidelines][https://github.com/bisq-network/style/issues)
- pick a descriptive title
- provide some meaningful PR description below
- create the PR
- in case you receive a "Change request" and/or a NACK, please react within 30 days. If not, we will close your PR and it can not be up for compensation.
- After addressing the change request, __please re-request a review!__ Otherwise we might miss your PR as we tend to only look at pull requests tagged with a "review required".
-->

Currently, the X-axis for the Price and Volume charts is not well designed. The labels are large, all the tick marks are the same. It is difficult to estimate to which tick mark the date label corresponds.  Here is a typical example for the DAY interval:
![Bisq-chartaxes-day_orig-2020-10-27](https://user-images.githubusercontent.com/142019/97351164-0c942e00-189a-11eb-9d90-efb66982e7c0.png)


The fix proposed here draws the labels more compact and the respective ticks stand-out. The chart view becomes cleaner and more visually intuitive. Above example, becomes:
![Bisq-chartaxes-day_new-2020-10-27](https://user-images.githubusercontent.com/142019/97351201-187ff000-189a-11eb-8c87-4bee28332fb1.png)

The labels take less horizontal space by splitting into 2-levels, using 24-hour time instead of AM/PM. The examples here show English locale, though localization for the month names is maintained, as before.

Of course, further customization is possible, but the above, I feel, provides an acceptable improvement to chart display, without deviating from the current UI or introducing new features.

Similar results are obtained with the other chart interval  values. Examples (10min, HOUR, WEEK, MOTH):
![Bisq-chartaxes-10min_new-2020-10-27](https://user-images.githubusercontent.com/142019/97348987-955d9a80-1897-11eb-9991-7abe3213aff8.png)
![Bisq-chartaxes-hour_new-2020-10-27](https://user-images.githubusercontent.com/142019/97349048-a27a8980-1897-11eb-8ea7-d4ca27d58e41.png)
![Bisq-chartaxes-week_new-2020-10-27](https://user-images.githubusercontent.com/142019/97349073-adcdb500-1897-11eb-9610-854ae7685803.png)
![Bisq-chartaxes-month_new-2020-10-27](https://user-images.githubusercontent.com/142019/97349093-b625f000-1897-11eb-8779-d9694558ad6a.png)

Bisq is a unique tool. It deserves better charts.
Hope you find this a small improvement in the right direction.

deusmax committed:
Cleaner formatting for trading charts date axes

Clean up date axes style for easy reading. Made date labels shorter,
with clear visual correspondance to relevant axes tick mark.
Tick marks with labels are now larger than those without.

Signed-off-by: Deus Max <deusmax@gmx.com>

ripcurlx commented:
@deusmax I think this is a good idea! Maybe we can get it even cleaner by getting rid of the day & month when year is selected.
![Bildschirmfoto 2020-10-29 um 10 41 27](https://user-images.githubusercontent.com/170962/97551622-7fed8b00-19d3-11eb-80ff-a3ccd11755f9.png)
And get rid of the day when month is selected.
![Bildschirmfoto 2020-10-29 um 10 41 12](https://user-images.githubusercontent.com/170962/97551703-9a276900-19d3-11eb-9d67-b71f76374d41.png)
What do you think?

deusmax commented:
@ripcurlx  Thanks. Agree these are good points and indeed the charts would be even better.
These changes you propose, may require to do some minor work on the chart logic. 
The code here applies changes only the chart's element configuration/formatting. It does not make any logical changes, even minor ones. In fact I explicitly avoided doing so.

I'll start working on the labels as you point out.... and get back.
Also, the Year-interval  charts don't really need to go all the way back to 1930 !!, do they ?

deusmax committed:
Added date format for each chart interval period.

Provide specific formats for each trade chart interval period, for added
precision and flexibility in formatting the X-axis. Covers suggestion in
Bisq PR#4715 for more compact date formats.

deusmax commented:
Implemented the suggestions by @ripcurlx .

**Yearly charts**, show only year.
![Bisq-chartaxes-year_new-2020-10-29](https://user-images.githubusercontent.com/142019/97642631-5cb3f180-1a4e-11eb-858b-7928404c6a45.png)

**Monthly charts,** got rid of the date and chopped the year.
![Bisq-chartaxes-month_new-2020-10-29](https://user-images.githubusercontent.com/142019/97642681-76553900-1a4e-11eb-96e0-8bc52eb65ba8.png)

ghubstan commented on desktop/src/main/java/bisq/desktop/main/market/trades/TradesChartsView.java:
```diff
@@ -479,11 +479,27 @@ private void layoutChart() {
+                case HOUR  :
+                case MINUTE_10: fmt = "HH:mm\ndd/MMM";
+                    break;
+                default:        // nothing here
```
You might want to reformat the style of the switch statement so `fmt = "..."` are on separate lines.  Not sure codacy will complain about this, but other devs might want the switch style to be consistent with the rest of the code base.
(This is the 1st time I've seen this style in Bisq;  I'm not saying there may be other examples in the code base.)

```
switch (model.tickUnit) {
                    case YEAR:
                        fmt = "yyyy";
                        break;
                    case MONTH:
                        fmt = "MMMyy";
                        break;
                    case WEEK:
                    case DAY:
                        fmt = "dd/MMM\nyyyy";
                        break;
                    case HOUR:
                    case MINUTE_10:
                        fmt = "HH:mm\ndd/MMM";
                        break;
                    default:        // nothing here
                }

```

ghubstan reviewed (commented):
I tested it, looks nice.  If I could (I'm not a JFX expert), I would center-align the labels where possible, e.g.,
```
29/Oct
 2020
```

But I would not block approval and merging without that.  
I will add a `Tested ACK` if the switch statement styling is changed.

deusmax committed:
Minor style formatting.

Fixed a switch statement formatting to conform with Bisq code style.

deusmax commented:
@ghubstan : pushed the formatting block changes.

Still looking into centering the labels.

ghubstan commented:
> @deusmax: 
>
> I will add a Tested ACK if the switch statement styling is changed.

> @ghubstan : pushed the formatting block changes.
> 
> Still looking into centering the labels.


Tested ACK.
